src/sounds.cpp

```diff
@@ -200,6 +202,8 @@ void sounds::process_sound_markers( player *p )
 
     for( const auto &sound_event_pair : sounds_since_last_turn ) {
         const int volume = sound_event_pair.second.volume * volume_multiplier;
+        const std::string sfx_id = sound_event_pair.second.id;
```

Comment:
Good point, I haven't had to write efficient code in too long. I hear you also made a string-int class like I'd always wanted to have one of, I might even look into using that. :)